pkgs/development/tools/git-series/default.nix

```diff
@@ -13,14 +13,24 @@ buildRustPackage rec {
     sha256 = "07mgq5h6r1gf3jflbv2khcz32bdazw7z1s8xcsafdarnm13ps014";
   };
 
-  depsSha256 = "1xypk9ck7znca0nqm61m5ngpz6q7c0wydlpwxq4mnkd1np27xn53";
+  cargoSha256 = "08fa8yn6chsb6fxhy4ks49z31kx4mhaybx7k6xw0zk9vz19hqjsw";
+
+  cargoDepsHook = ''
+    (
+      cd ${name}-src
+      # see https://github.com/git-series/git-series/pull/56
+      patch -p1 < ${fetchpatch {
+        url = "https://github.com/Mic92/git-series/commit/3aa30a47d74ebf90b444dccdf8c153f07f119483.patch";
+        sha256 = "06v8br9skvy75kcw2zgbswxyk82sqzc8smkbqpzmivxlc2i9rnh0";
+      }}
```

Comment:
With this trick state Cargo.lock files can be fixed.

Reply:
nice. maybe it would be less obscure if the cargo deps where they own derivation:
```nix
{
deps = fetchcargo {
  src = ./Cargo.lock;
  sha256 = "08fa8yn6chsb6fxhy4ks49z31kx4mhaybx7k6xw0zk9vz19hqjsw";
  postUnpack = " # patch all the things ";
};
}
```

Reply:
We could also move `patchPhase` before cargo runs.

Reply:
yes good idea, the patchPhase should operate on what is in the src, so it makes sense to run before the "fetchcargo" phase.

Reply:
actually it's going to be tricky because `src + "/Cargo.lock"` is still going to contain the original

Reply:
Maybe I didn't understand but if the patch phase comes before the cargo fetch then fixing/changing a `Cargo.lock` file would be like any other patch. I guess that would be a clean solution?

Reply:
This issue is how it's currently composed. Right now the same src is passed to both fetchDeps() (aka cargofetch) and the parent stdenv.mkDerivation(). The phases we are talking about are only happening in the context of the stdenv.mkDerivation realization.